cdap-app-fabric/src/main/java/io/cdap/cdap/metadata/DatasetFieldLineageSummary.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.metadata;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.proto.id.DatasetId;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * The summary about all the field level lineage information about all fields in a dataset.
+ */
+public class DatasetFieldLineageSummary {
+  private final Constants.FieldLineage.Direction direction;
+  private final long startTs;
+  private final long endTs;
+  private final DatasetId entityId;
```

Comment:
wouldn't it make more sense to name this dataset instead of entity?

Reply:
Or was it named this way on purpose to be able to use for something other than datasets?

Reply:
There is no specific reason for this naming, it is just following the design doc response. I can verify and update.

Reply:
I think we called it entityId to keep things general enough that we could potentially use it for other kinds of entities too.

Reply:
👍 I will use entityId as the serialized name so if we support other entities, we do not need to modify the response.